### PR TITLE
Update comment counts

### PR DESCRIPTION
**Pull request by unDemian**
**Built on top of https://github.com/Automattic/wp-calypso/pull/40720**

#### Changes proposed in this Pull Request

Update realtime comment counts
- In following view http://calypso.localhost:3000/read
- In feed view http://calypso.localhost:3000/read/feeds/<feedId> you can get there by clicking the author name in the following view
- In full post view for both feed and blog http://calypso.localhost:3000/read/blogs/<id>/posts/<id> by clicking the post title in a feed/conversation

#### Testing instructions
Example https://d.pr/v/euB0QQ

- Open all the above pages
- Apply patch D41517
- Sandbox one of your blogs, follow it and start a conversation (multiple comments and replies)
- Add a comment to one of your posts on the frontend, the comment count should increase in calypso
- Try the same after hard refreshing the section

### CODE REVIEW

**Comment by hewsut** [comment]
What about bumping the count in the post instead of making a separate pathway? Something like this (not fully tested -- there's a bit of odd what I presume to be placeholder driven flutter in the window posting):

```
diff --git a/client/state/comments/actions.js b/client/state/comments/actions.js
index 6ef35729ef..026a1ff093 100644
--- a/client/state/comments/actions.js
+++ b/client/state/comments/actions.js
@@ -6,6 +6,7 @@ import {
        COMMENT_COUNTS_REQUEST,
        COMMENT_REQUEST,
        COMMENTS_CHANGE_STATUS,
+       COMMENTS_COUNT_INCREMENT,
        COMMENTS_DELETE,
        COMMENTS_EDIT,
        COMMENTS_LIKE,
@@ -156,6 +157,19 @@ export const requestCommentCounts = ( siteId, postId ) => ( {
        postId,
 } );
 
+/**
+ * Creates an action that increments the comment count for a given post.
+ *
+ * @param {number} siteId Site identifier
+ * @param {number} [postId] Post identifier
+ * @returns {object} Action that increments comment count for a post.
+ */
+export const incrementCommentCount = ( siteId, postId ) => ( {
+       type: COMMENTS_COUNT_INCREMENT,
+       siteId,
+       postId,
+} );
+
 /**
  * Creates an action that permanently deletes a comment
  * or removes a comment placeholder from the state
diff --git a/client/state/data-layer/wpcom/sites/utils.js b/client/state/data-layer/wpcom/sites/utils.js
index 04d4e02cb8..4bb9ef8d2b 100644
--- a/client/state/data-layer/wpcom/sites/utils.js
+++ b/client/state/data-layer/wpcom/sites/utils.js
@@ -106,8 +106,6 @@ export const updatePlaceholderComment = (
                                },
                        },
                },
-               // increment comments count
-               { type: COMMENTS_COUNT_INCREMENT, siteId, postId },
        ];
 
        if ( !! refreshCommentListQuery ) {
diff --git a/client/state/reader/posts/reducer.js b/client/state/reader/posts/reducer.js
index f6d6f84bc7..9531d8cb59 100644
--- a/client/state/reader/posts/reducer.js
+++ b/client/state/reader/posts/reducer.js
@@ -2,11 +2,12 @@
 /**
  * External dependencies
  */
-import { keyBy, get } from 'lodash';
+import { findKey, get, keyBy } from 'lodash';
 
 /**
  * Internal dependencies
  */
+import { COMMENTS_COUNT_INCREMENT } from 'state/action-types';
 import { READER_POSTS_RECEIVE, READER_POST_SEEN } from 'state/reader/action-types';
 import { combineReducers } from 'state/utils';
 
@@ -22,7 +23,29 @@ export function items( state = {}, action ) {
                case READER_POSTS_RECEIVE:
                        const posts = action.posts || action.payload.posts;
                        return { ...state, ...keyBy( posts, 'global_ID' ) };
+
+               case COMMENTS_COUNT_INCREMENT: {
+                       const globalId = findKey( state, ( { site_ID, ID } ) => {
+                               return site_ID === action.siteId && ID === action.postId;
+                       } );
+
+                       if ( ! globalId ) {
+                               return state;
+                       }
+
+                       return {
+                               ...state,
+                               [ globalId ]: {
+                                       ...state[ globalId ],
+                                       discussion: {
+                                               ...state[ globalId ].discussion,
+                                               comment_count: state[ globalId ].discussion.comment_count + 1,
+                                       },
+                               },
+                       };
+               }
        }
+
        return state;
 }
```

**Comment by hewsut** [comment]
Woof, the "flake" is the flutter between the `post-comment-list.jsx` way of calculating -- which takes into account the existing tree and what's pending -- and that the `COMMENTS_COUNT_INCREMENT` dispatch happens after the roundtrip to the serverside.

tl;dr: we need to extract the fancy logic that gives `actualCommentCount` to somewhere else, so we can use it in `ReaderPostActions` too.

(on the bright side....we'd be fixing this as `master` doesn't update at all!)

**Comment by unDemian** [comment]
This is still work in progress, at first I thought I can just increment totals but as it turns out on hard refreshes it won't work so I added the global_ID on the backend to make it work, I guess updating the inline reader post would make more sense now that we have the global id, I'll have to take a look comments and comments/counts is just so much fun when combined with reader's posts.

**Comment by unDemian** [comment]
I'm not quite sure I can reproduce the flakiness you mentioned, could you provide a quick video recording if possible. I pushed some updates that would contain the comments count to the reader concern.

**Comment by unDemian** [comment]
Closed in favor of using lasagna.js
